middleware/TelemetryHandler.md

```diff
@@ -2,17 +2,47 @@
-- Add `SdkVersion` request header to each request to identify the language and version of the client SDK library.  The product identifier should be of the format `graph-{lang}`. The `SdkVersion` version can contain multiple values, or appear multiple times in a request. Values should be comma delimited as normal header lists are if the `SdkVersion` contains multiple values. e.g.
 
-    `SdkVersion: MyApp/1.0, spfx/1.0, graph-javascript-1.0`
-- If available, add a the `featureUsage` value contained in the [RequestContext](../middleware/RequestContext.md) as key/value pair in the comment after the product. e.g.
+- Add `SdkVersion` request header to each request to identify the language and version of the client SDK library(s). The product identifier should be of the format `graph-{lang}/{majorVersion}.{minorVersion}.{buildVersion}` for client libraries. If the client SDK library has more than one component i.e. core, v1.0/beta service model library, then the product identifier should capture them as follows based on usage :
 
-    `SdkVersion: graph-dotnet/1.0 (featureUsage=0f)`
+    | Client SDK Library Component | Template | Example |
+    | --- | --- | --- |
+    | **Core** | `graph-{lang}-core/{major}.{minor}.{build}` | `graph-dotnet-core/1.16.0` |
+    | **v1.0 Service Model** | `graph-{lang}/{major}.{minor}.{build}` | `graph-dotnet/1.16.0` |
+    | **Beta Service Model** | `graph-{lang}-beta/{major}.{minor}.{build}` | `graph-dotnet-beta/1.16.0` |
+    
+    The `SdkVersion` version can contain multiple values, or appear multiple times in a request. Values should be comma delimited as normal header lists are if the `SdkVersion` contains multiple values. e.g.
+
+    ```
+    SdkVersion: MyApp/1.0, graph-dotnet-core/1.16.0 (featureUsage=0f), graph-dotnet/1.15.0
+    ```
+
+- If available, add a the `featureUsage` value contained in the [RequestContext](../middleware/RequestContext.md) to the `SDKVersion` header as key/value pair in the comment after the product identifier. e.g.
+
+    `SdkVersion: graph-javascript/1.0.0 (featureUsage=0f)`
+
+- Add `HostOS` request header to each request to help us identify the OS on which our client SDK is running on. e.g.
+
+    `HostOS: Microsoft Windows 10.0.18362`
+
+- Add `RuntimeEnvironment` request header to capture the runtime framework on which the client SDK is running on. Ideally, we should capture the runtime environment in the form of `RuntimeEnvironment: Name/Version`. e.g.
+
+    - `RuntimeEnvironment: .NETFramework/1.1` for .NET.
+    - `RuntimeEnvironment: Node/1.1` for JavaScript.
+    - `RuntimeEnvironment: JRE/1.1` for Java.
+
+#### Ideal Metadata Capture
+```
+SdkVersion: graph-dotnet-beta/0.6.0-preview, graph-dotnet-core/1.16.0 (featureUsage=0f)
+client-request-id: fdae6861-5916-486d-93d9-9129160b2d79
+HostOS: Microsoft Windows 10.0.18362
+RuntimeEnvironment: .NETFramework/4.7.2
```

Comment:
I'm still not convinced we should introduce two new headers.  
Why not? 

    SdkVersion: graph-dotnet-beta/0.6.0-preview, graph-dotnet-core/1.16.0 (featureUsage=0f),  .NETFramework/1.1, Windows/10.0.18362

Reply:
Should the community project be adding to that header?

Reply:
It certainly can.  Spfx, the Graph Toolkit and Postman collections do, so why not.

Reply:
Won't this abstract the environment values being put in the header i.e. hostOS and runtimeEnvironment. How about we introduce an extra header e.g. `HostEnvironment`, to hold the two values (hostOS and runtimeEnvironment).

On second thought, since the headers are meant to be used by us, we can control how we siphon info out of them, thus your suggestion is 100% valid and keeps everything contained in one header.